### PR TITLE
KP/PP Classification Tool Updates

### PR DESCRIPTION
**Pull request by dbaluku**
https://trello.com/c/wvIccnHJ

### CODE REVIEW

**Comment by dbaluku** [comment]
Updates on the KP/PP Classification Tool is:

-  Automatic checking of options in the key population Classification Section basing on the client information fed in.

File: omod/src/main/webapp/resources/htmlforms/KP_PP_Classification_Tool.xml [hunk 1]
```diff
@@ -65,20 +227,31 @@
...
                 </thead>
-                <tbody>
+                <tbody class="triggers">
+                    <tr>
+                        <td>
+                            Encounter Date:
+                            <encounterDate default="today" disallowMultipleEncountersOnDate="block"/>
+                        </td>
+                        <td>
+                            <label><span class="required">*</span>Assesed by:</label>
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/htmlforms/KP_PP_Classification_Tool.xml]: @balukujr Correct "Assesed" to "Assessed"

**Comment by ssmusoke** [comment]
@balukujr 

1. For questions #2 and #4 when I select other without entering a value the validation is not triggere, so need to add validations for this 

2. TG - when I select Refuse to Answer for Sex at Birth, then this should not be selected 

3. Final Classification - only the KP/PP classifications selected should be available for selection, so the client should not have a final classification to which they do not belong. Maybe disable all of them on the form load, and only enable those that are selected

**Comment by dbaluku** [comment]
Issues above have been worked @ssmusoke  please review the updated form

File: omod/src/main/webapp/resources/htmlforms/KP_PP_Classification_Tool.xml [hunk 2]
```diff
@@ -43,11 +43,280 @@
...
         }
-        
+
 
     </style>
+    <script type="text/javascript">
+        if(jQuery){
+          jQuery(document).ready(function(){
+
+                 jq("#msm_final").find(":checkbox").attr("disabled","disabled");
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/htmlforms/KP_PP_Classification_Tool.xml]: @balukujr Can you also ensure that the KP classifications, not just the final ones are also disabled when the page loads. For example I can be Male at Birth, and think of myself as Male but someone can select TG.

File: omod/src/main/webapp/resources/htmlforms/KP_PP_Classification_Tool.xml [hunk 3]
```diff
@@ -43,11 +43,284 @@
...
         width: 93%;
         }
-        
+
 
     </style>
+    <script type="text/javascript">
+        if(jQuery){
+          jQuery(document).ready(function(){
+                <ifMode mode="ENTER">
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/htmlforms/KP_PP_Classification_Tool.xml]: 1. Can you also have the final classification be disabled this way on both Enter and Edit mode.

2. When in edit mode, there is need to enable any of the checkboxes which have values selected as they will be loaded with values and already checked

File: omod/src/main/webapp/resources/htmlforms/KP_PP_Classification_Tool.xml [hunk 4]
```diff
@@ -43,11 +43,284 @@
...
+                                getField('refused_to_answer.error').html('Required Field').show();
+                                return false;
+                            }else{
+                            setValue('refused_to_answer.error', '');
+                            }
+
+                             <!-- if the other is checked and no specification made -->
+                        if (getValue('your-gender.value')==="5622") {
+                            if (getValue('other-specify.value') === '') {
+                                getField('other-specify.error').html('This field is required ').show();
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/htmlforms/KP_PP_Classification_Tool.xml]: Change the error - Please specify the other gender you consider yourself

File: omod/src/main/webapp/resources/htmlforms/KP_PP_Classification_Tool.xml [hunk 5]
```diff
@@ -43,11 +43,284 @@
...
+                            }
+                            else{
+                            <!--setValue('other-specify.error', '');-->
+                             return jq("#htmlform").valid();
+                            }
+                        }
+
+                         if (getValue('trigger.value')==="5622") {
+                            if (getValue('other-occ.value') === '') {
+                                getField('other-occ.error').html('This field is required ').show();
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/htmlforms/KP_PP_Classification_Tool.xml]: Change message to "Please specify the other occupation"

**Comment by dbaluku** [comment]
@ssmusoke  Please review the new updates made..

**Comment by ssmusoke** [comment]
1. Validate to ensure that a dominant category is selected 

2. When editing enable the available dominant fields 

3. When None is selected for dominant classification all other fields should be unchecked 

4. In occupation when a user selects none, then uncheck all other occupations and make them read only

**Pull request merged**